### PR TITLE
Implement paper-checkbox

### PR DESCRIPTION
**Pull request by notwaldorf**
Hiya @morethanreal! This shooooould be the same as `paper-radio-button`, with the review comments from that PR already applied. Please take a look.

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  },
+  "license": "MIT",
+  "homepage": "https://github.com/PolymerElements/paper-checkbox",
+  "ignore": [
+  ],
+  "devDependencies": {
+    "webcomponentsjs": "webcomponents/webcomponentsjs#~0.6.1",
+    "web-component-tester": "Polymer/web-component-tester#^2.2.3",
+    "iron-doc-viewer": "PolymerElements/iron-doc-viewer#master",
+    "layout": "Polymer/layout#master",
```
Comment (morethanreal) [inline bower.json]: Please update to use layout in the new `PolymerElements/paper-styles`

File: bower.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    "test-fixture": "polymer/test-fixture#master",
+    "core-focusable": "polymer/core-focusable#0.8-preview"
+  },
+  "dependencies": {
+    "polymer": "polymer/polymer#0.8-preview",
+    "paper-ripple": "Polymer/paper-ripple#0.8-preview",
+    "paper-styles": "PolymerLabs/paper-styles#master"
```
Comment (morethanreal) [inline bower.json]: ditto

File: bower.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  "devDependencies": {
+    "webcomponentsjs": "webcomponents/webcomponentsjs#~0.6.1",
+    "web-component-tester": "Polymer/web-component-tester#^2.2.3",
+    "iron-doc-viewer": "PolymerElements/iron-doc-viewer#master",
+    "layout": "Polymer/layout#master",
+    "test-fixture": "polymer/test-fixture#master",
+    "core-focusable": "polymer/core-focusable#0.8-preview"
+  },
+  "dependencies": {
+    "polymer": "polymer/polymer#0.8-preview",
```
Comment (morethanreal) [inline bower.json]: update to v0.8.0-rc.4

File: demo/index.html [hunk 4]
```diff
@@ -0,0 +1,88 @@
+<!doctype html>
+<!--
+Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
```
Comment (morethanreal) [inline demo/index.html]: indent this block (see PolymerLabs/seed-element)

File: demo/index.html [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  <script src="../../webcomponentsjs/webcomponents-lite.js"></script>
+  <link rel="import" href="../../layout/layout.html">
+  <link rel="import" href="../paper-checkbox.html">
+
+  <link rel="stylesheet" href="../../paper-styles/demo.css">
+
+  <style is="x-style">
+    paper-checkbox.blue {
+      --paper-checkbox-unchecked-color: red;
+      --paper-checkbox-unchecked-ink-color: red;
```
Comment (morethanreal) [inline demo/index.html]: use colors from `default-theme.html`

File: index.html [hunk 6]
```diff
@@ -0,0 +1,35 @@
+<!doctype html>
+<!--
+Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
```
Comment (morethanreal) [inline index.html]: ditto

File: paper-checkbox.html [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+    --paper-checkbox-checked-color: #009688;
+    --paper-checkbox-checked-ink-color: #009688;
+  }
+</style>
+
+@group Paper Elements
+@class paper-checkbox
+-->
+
+<link href="../polymer/polymer.html" rel="import">
```
Comment (morethanreal) [inline paper-checkbox.html]: `<link rel="import">`

File: paper-checkbox.html [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+<link rel="import" href="../paper-ripple/paper-ripple.html">
+<link rel="import" href="../paper-styles/x-theme.html">
```
Comment (morethanreal) [inline paper-checkbox.html]: ditto re: PolymerElements/paper-styles

File: paper-checkbox.html [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+    onMouseDown: function() {
+      if (this.disabled) {
+        return;
+      }
+
+      var old = this.checked;
+      this.checked = !this.checked;
+
+      if (this.checked !== old) {
+        this.fire('core-change');
```
Comment (morethanreal) [inline paper-checkbox.html]: wonder if this should be `iron-change` now? @cdata

File: demo/index.html (hunk 4 above)
Comment (notwaldorf) [inline demo/index.html]: Just to double check, you mean the `<head>` bit, not the license, right?

File: demo/index.html [hunk 10]
```diff
@@ -0,0 +1,89 @@
+<!doctype html>
+<!--
+@license
+Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
```
Comment (morethanreal) [inline demo/index.html]: I do mean the license. See PolymerLabs/seed-element

**Comment by notwaldorf** [comment]
Alright. I've squashed this and teleported the history and I think I didn't break any extra things. :neutral_face:

File: bower.json [hunk 11]
```diff
@@ -1,16 +1,34 @@
...
+    "webcomponentsjs": "webcomponents/webcomponentsjs#~0.6.1",
+    "web-component-tester": "Polymer/web-component-tester#^2.2.3",
+    "iron-doc-viewer": "PolymerElements/iron-doc-viewer#master",
+    "test-fixture": "polymer/test-fixture#master",
+    "core-focusable": "polymer/core-focusable#0.8-preview"
+  },
+  "dependencies": {
+    "polymer": "polymer/polymer#v0.8.0-rc.4",
+    "paper-ripple": "Polymer/paper-ripple#0.8-preview",
+    "paper-styles": "PolymerElements/paper-styles#master"
```
Comment (morethanreal) [inline bower.json]: Use `^0.8.0`?

File: bower.json [hunk 12]
```diff
@@ -1,16 +1,34 @@
...
-    "externs/**",
-    ".gitignore"
   ],
   "devDependencies": {
-    "font-roboto": "Polymer/font-roboto#~0.5.4",
-    "core-label": "Polymer/core-label#~0.5.4"
+    "webcomponentsjs": "webcomponents/webcomponentsjs#~0.6.1",
+    "web-component-tester": "Polymer/web-component-tester#^2.2.3",
+    "iron-doc-viewer": "PolymerElements/iron-doc-viewer#master",
+    "test-fixture": "polymer/test-fixture#master",
```
Comment (morethanreal) [inline bower.json]: `^0.8.0`?

File: demo/index.html [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+-->
+<html>
+  <head>
+    <title>paper-checkbox</title>
```
Comment (morethanreal) [inline demo/index.html]: paper-checkbox demo

File: paper-checkbox.html [hunk 14]
```diff
@@ -1,102 +1,177 @@
...
+      this.checked = !this.checked;
+
+      if (this.checked !== old) {
+        this.fire('iron-change');
+      }
+    },
+
+    _checkedChanged: function() {
+      this.setAttribute('aria-checked', this.checked ? 'true' : 'false');
+      this.fire('core-change');
```
Comment (morethanreal) [inline paper-checkbox.html]: `iron-change` here too?

**Comment by morethanreal** [comment]
lgtm w/ nit

**Pull request merged**